File: javascript/misc-secondLargestInArr.js
var secondLargest = function(arr) {
  let largest = 0;
  let second = 0;

  for (let i = 0; i < arr.length; i++) {
    if (arr[i] > largest) {
      second = largest;
      largest = arr[i];
    } else if (arr[i] > second) {
      second = arr[i];
    }
  }

  return secondLargest;
};

File: python/misc-secondLargestInArr.py


class Solution(object):
    def secondLargest(nums):
    largest = nums[0]
    second = nums[0]

    for i in range(0, len(nums)):
        if nums[i] > largest:
            second = largest
            largest = nums[i]
        elif nums[i] > second & nums[i] != largest:
            second = nums[i]

    print(second)
